### DISCUSSION TITLE
Request deflate support in mojolicious

### DISCUSSION THREAD

**Discussion by rmallah** [mojolicious/mojo#1959] [Ideas]
In a mojo app i had to do this:

```

    $self->hook( before_dispatch => sub {
      my ($c) = @_;
      my $encoding = $c->req->headers->header('Content-Encoding');
      if ( $encoding =~ /gzip/) {
        $c->req->body( Compress::Zlib::memGunzip ( $c->req->body)  );
      }
     });

```

is it being done transparently already and i am not knowing.
i know Mojo::UserAgent does it automatically when fetching
resources of the net.

**Comment by kraih** [comment]
It's not really a thing for incoming requests on the backend.
